server/balancer.go

```
@@ -0,0 +1,316 @@
+package server
+
+import (
+	"github.com/golang/protobuf/proto"
+	"github.com/juju/errors"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pingcap/kvproto/pkg/raftpb"
+)
+
+// Operator is the interface to do some operations.
+type Operator interface {
```

I think we should return the error to cancel this region balance too.